### ISSUE TITLE
Issue with nested relpaths on Windows

### ISSUE THREAD

**Issue by tjelvar-olsson** [jic-dtool/dtoolcore#13]
The test below currently fails on Windows.

```python
def test_basic_workflow_with_nested_relpath(tmp_dir_fixture):  # NOQA

    from dtoolcore import ProtoDataSet, generate_admin_metadata
    from dtoolcore import DataSet
    from dtoolcore.utils import generate_identifier
    from dtoolcore.storagebroker import DiskStorageBroker

    name = "my_dataset"
    admin_metadata = generate_admin_metadata(name)
    dest_uri = DiskStorageBroker.generate_uri(
        name=name,
        uuid=admin_metadata["uuid"],
        base_uri=tmp_dir_fixture)

    sample_data_path = os.path.join(TEST_SAMPLE_DATA)
    local_file_path = os.path.join(sample_data_path, 'tiny.png')
    dest_relpath = os.path.join("subdir", "tiny.png")

    # Create a minimal dataset
    proto_dataset = ProtoDataSet(
        uri=dest_uri,
        admin_metadata=admin_metadata,
        config_path=None)
    proto_dataset.create()
    handle = proto_dataset.put_item(local_file_path, dest_relpath)

    proto_dataset.freeze()

    # Read in a dataset
    dataset = DataSet.from_uri(dest_uri)

    expected_identifier = generate_identifier(handle)
    assert expected_identifier in dataset.identifiers
    assert len(dataset.identifiers) == 1
```

**Comment by tjelvar-olsson** [comment]
On Windows the dtoolcore.storagebroker.DiskStorageBroker.iter_item_handles converts relpaths to Unix style relpaths.

There is also logic to convert Unix paths to Windows paths when listing dataset URIs and for converting Windows paths to Unix paths when generating dataset URIs.

However, there is no logic for dealing with nested relpaths on Windows in:

- ProtoDataSet.put_item
- DiskStorageBroker.put_item
- DiskStorageBroker.get_item_abspath

What does relpath mean on Windows?

I think it should mean a Windows relpath, e.g. "subdir\\my_data.txt".
This makes semantic sense and distinguishes it from our concept of a handle, which will continue to be a Unix style relpath.

Speaking of the Unix style relpath, we should probably add some code to verify and sanitise these. I.e.

1. Raise errors on any relpath that starts with ../
2. Convert ./my/file.txt to my/file.txt

If we agree that relpath means Windows style relpath on Windows I think we need to:

1. Update DiskStorageBroker.put_item to create Unix style handle from the Windows relpath and return this instead of just passing through the relpath that was passed in.

2. Add some logic to DiskStorageBroker.get_item_abspath to convert Unix style relpath to Windows style relpath

3. Additionally I think it might be good to add verification and sanitisation of handles returned from ProtoDataSet.put_item.

**Comment by tjelvar-olsson** [comment]
Following on from discussion with @mrmh2 

1. Keep windows specific logic in DiskStorageBroker, this includes DiskStorage.put_item and DiskStorage.get_item_abspath (and possibly others)
2. Rename argument in put_item from "relpath" to "handle" and explain what this is (this is currently a subtle defect in the sematics)

**Comment by tjelvar-olsson** [comment]
Fixed in 5110b32
